### PR TITLE
disable jemalloc on executables for ios targets

### PR DESCRIPTION
**Pull request by kali**
This is a (temporary ?) workaround for issue #45262

### CODE REVIEW

**Comment by carols10cents** [comment]
Review ping for you @pnkfelix !

File: src/librustc_back/target/apple_ios_base.rs [hunk 1]
```diff
@@ -99,6 +99,7 @@ pub fn opts(arch: Arch) -> Result<TargetOptions, String> {
         executables: true,
         pre_link_args,
         has_elf_tls: false,
+        exe_allocation_crate: None,
```
Comment (pnkfelix) [inline src/librustc_back/target/apple_ios_base.rs]: @kali can you add a comment noting that this new line can be removed after Rust upgrades to a sufficiently new version of jemalloc (and mention #45262 in the comment)?

Thanks!

**Comment by pnkfelix** [comment]
r=me once the comment I requested above has been added.

**Comment by sfackler** [comment]
@alexcrichton thoughts on backporting to beta? Seems like it'd be nice to fix iOS bins earlier rather than later and it should be low risk.

**Comment by alexcrichton** [comment]
Sure!

**Comment by alexcrichton** [comment]
Looks like we forgot to backport to 1.23.0 (sorry about that!) so removing the beta tags.

**Pull request merged**